From 9ef05281e5d01dffdf75a4ae5c1aee7cbd7aaa73 Mon Sep 17 00:00:00 2001
Message-Id: <9ef05281e5d01dffdf75a4ae5c1aee7cbd7aaa73.1648734868.git.geliang.tang@suse.com>
From: Jianlin Lv <Jianlin.Lv@arm.com>
Date: Thu, 18 Mar 2021 10:48:51 +0800
Subject: [PATCH] bpf: Remove insn_buf[] declaration in inner block
Git-commit: 9ef05281e5d01dffdf75a4ae5c1aee7cbd7aaa73
Patch-mainline: v5.13-rc1
References: bsn#127
Modified-by-SEL: Yes, refreshed due to context changes


Two insn_buf[16] variables are declared in the function which acts on
function scope and block scope respectively. The statement in the inner
block is redundant, so remove it.

Signed-off-by: Jianlin Lv <Jianlin.Lv@arm.com>
Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>
Link: https://lore.kernel.org/bpf/20210318024851.49693-1-Jianlin.Lv@arm.com
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 kernel/bpf/verifier.c |    1 -
 1 file changed, 1 deletion(-)

--- a/kernel/bpf/verifier.c
+++ b/kernel/bpf/verifier.c
@@ -11822,7 +11822,6 @@ static int fixup_bpf_calls(struct bpf_ve
 		    insn->code == (BPF_ALU64 | BPF_SUB | BPF_X)) {
 			const u8 code_add = BPF_ALU64 | BPF_ADD | BPF_X;
 			const u8 code_sub = BPF_ALU64 | BPF_SUB | BPF_X;
-			struct bpf_insn insn_buf[16];
 			struct bpf_insn *patch = &insn_buf[0];
 			bool issrc, isneg, isimm;
 			u32 off_reg;
